### ISSUE TITLE
Include `extern` information in the XML output

### ISSUE THREAD

**Issue by michaeljones** [doxygen/doxygen#10163]
**Describe the bug**
The XML output does not include indication of when a function or variable is `extern`.

**Expected behavior**
The `memberdef` entries include an `extern` or `external` attribute which either defaults to `no` or is only present when `yes`. 

**Screenshots**
```diff
-      <memberdef kind="function" id="example_8hpp_1a267724ea80f41018f14782678d097888" prot="public" static="no" const="no" explicit="no" inline="no" virt="non-virtual">
+      <memberdef kind="function" id="example_8hpp_1a267724ea80f41018f14782678d097888" prot="public" static="no" const="no" explicit="no" inline="no" virt="non-virtual" extern="yes">
```

**To Reproduce**
[extern.zip](https://github.com/doxygen/doxygen/files/11930010/extern.zip)

**Version**
doxygen: 1.9.6 
doxygen: 1.9.7
Binaries from doxygen website.
```
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.2 LTS
Release:        22.04
Codename:       jammy
```

**Additional context**
This was done for the sqlite output in https://github.com/doxygen/doxygen/pull/8533 and used the name `extern` for the field so it seems reasonable to copy that. I am not experienced with the doxygen code base but it seems like the `extern` entry is added unconditionally. 

This would help Breathe which is a project that relies on the XML output: https://github.com/breathe-doc/breathe/issues/527 and it would help [Docleaf](https://github.com/docleaf-labs/docleaf) which takes a similar approach. When processing the Zephyr RTOS code base we end up with clashes between the original declarations of functions and `extern` declarations of the same functions in other files.

**Comment by michaeljones** [comment]
Hmm, I might be wrong (my C++ is rusty and my C is almost non-existent) but it seems like `extern` isn't being picked up properly. Adding:

```c
  if (md->isExternal())
  {
    t << " extern=\"yes\"";
  }

```
Here: https://github.com/doxygen/doxygen/blob/906571f37afbb3e8ebb5be3a88a35c239adefa73/src/xmlgen.cpp#L644-L650

Doesn't seem to output `extern` into the xml for function `<member>` entries. It does work for the variable entry in my very basic test case of:

```c
/*! An extern function */
extern void example_function(int arg1, bool arg2);

/*! An extern variable */
extern int var;
```

I'm not familiar with the code base but I'll attempt to poke around a little when I have time.

**Comment by michaeljones** [comment]
Ah, I have found that the `explicitExternal` information was not being passed from the `Entry` to the `MemberDef` in `addGlobalFunction`. I have attempted to address that and add the `extern` attribute to the xml output here: https://github.com/doxygen/doxygen/pull/10164

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
